src/tcp_sock.rs

```diff
@@ -324,14 +354,14 @@ impl Drop for Inner {
 }
 
 /// Serialize given value and write to a buffer with 4 byte length header.
-fn serialize_with_len<T: Serialize>(value: T) -> ::Res<Vec<u8>> {
-    let mut data = Cursor::new(Vec::with_capacity(mem::size_of::<u32>()));
+fn serialize_with_len<T: Serialize>(value: T, enc_ctx: &EncryptContext) -> ::Res<Vec<u8>> {
+    let encrypted = enc_ctx.encrypt(&value)?;
 
-    let _ = data.write_u32::<LittleEndian>(0);
-    serialise_into(&value, &mut data)?;
-    let len = data.position() - mem::size_of::<u32>() as u64;
-    data.set_position(0);
-    data.write_u32::<LittleEndian>(len as u32)?;
+    let mut data = Cursor::new(Vec::with_capacity(mem::size_of::<u32>() + encrypted.len()));
+    let _ = data.write_u32::<LittleEndian>(encrypted.len() as u32);
```

Comment:
I think the task specifically asks to not write plaintexts - pls read the task again if you missed a few points from there.

Reply:
https://github.com/maidsafe/socket-collection/issues/10
```
Investigate that encrypting the payload size header (plaintext size of 4 bytes representing a u32) always produces a consistent length of ciphertext. Otherwise it'll be quite hard to decrypt the payload size header at the recipient.
```
I haven't done proper research on how to encrypt the length header so that result would always be a constant size - otherwise we won't be able to decrypt it. Encryption itself will yield a lot of changes and I'm taking this issue step by step